--- django/db/backends/postgresql/base.py.orig
+++ django/db/backends/postgresql/base.py
@@ -86,7 +86,21 @@ class DatabaseWrapper(local):
         global postgres_version
         if not postgres_version:
             cursor.execute("SELECT version()")
-            postgres_version = [int(val) for val in cursor.fetchone()[0].split()[1].split('.')]        
+            version_parts = cursor.fetchone()[0].split()[1].split('.')
+            postgres_version = []
+            for part in version_parts:
+                if part.isdigit():
+                    postgres_version.append(int(part))
+                else:
+                    digit = ""
+                    for c in part:
+                        if c.isdigit():
+                            digit += c
+                        else:
+                            break
+                    if digit != "":
+                        postgres_version.append(int(digit))
+                    break
         if settings.DEBUG:
             return util.CursorDebugWrapper(cursor, self)
         return cursor
