Subject: Documentation not building on Sphinx 1.2.1
Origin: https://github.com/django/django/commit/e1d18b9d2e8ac292940f070b0a8cb9733756acd9#diff-b6a99c1d3137306aa674102ff23eafa4R249
Bug: https://code.djangoproject.com/ticket/21869
Forwarded: not-needed

--- a/docs/_ext/djangodocs.py
+++ b/docs/_ext/djangodocs.py
@@ -101,11 +101,13 @@
         self.compact_p = self.context.pop()
         self.body.append('</table>\n')
 
-    # <big>? Really?
     def visit_desc_parameterlist(self, node):
-        self.body.append('(')
+        self.body.append('(')  # by default sphinx puts <big> around the "("
         self.first_param = 1
+        self.optional_param_level = 0
         self.param_separator = node.child_text_separator
+        self.required_params_left = sum([isinstance(c, addnodes.desc_parameter)
+                                         for c in node.children])
 
     def depart_desc_parameterlist(self, node):
         self.body.append(')')
